src/pybind/mgr/cephadm/module.py

```diff
@@ -643,22 +638,39 @@ def _run_cephadm(self, host, entity, command, args,
             final_args += args
 
             if self.mode == 'root':
-                self.log.debug('args: %s' % final_args)
+                self.log.debug('args: %s' % (' '.join(final_args)))
                 self.log.debug('stdin: %s' % stdin)
                 script = 'injected_argv = ' + json.dumps(final_args) + '\n'
                 if stdin:
                     script += 'injected_stdin = ' + json.dumps(stdin) + '\n'
                 script += self._cephadm
-                out, err, code = remoto.process.check(
-                    conn,
-                    ['/usr/bin/python', '-u'],
-                    stdin=script.encode('utf-8'))
+                first_err = None
+                for python in PYTHONS:
+                    try:
+                        out, err, code = remoto.process.check(
+                            conn,
+                            [python, '-u'],
+                            stdin=script.encode('utf-8'))
+                    except RuntimeError as e:
```

Comment:
is there a way to explicitly check for the missing python interpreter here? I'm a bit nervous to run _all_ commands twice in case some error happens, just because we cannot distinguish the errors.

Reply:
Updated to use a remote function (yay, remoto!) to identify which python to use.